hips/hip-9997.md

```diff
@@ -0,0 +1,197 @@
+4. Chart names == OCI reference basenames
+5. Cache is removed
+6. `helm chart` is removed, integrated into rest of CLI
+7. Experimental until clear messaging from OCI
+
+### 1. Implement `Getter` and introduce `Pusher`
+
+As of yet, the code related to registry support has been written standalone (see [internal/experimental/registry](https://github.com/helm/helm/tree/master/internal/experimental/registry)).
+
+The act of downloading a chart from an OCI registry should mimic what is already possible using downloader plugins. Using the protocol prefix `oci://` in any place where chart repos are referenced should work just by implementing a `Getter` for OCI.
+
+Currently there is no Helm-specific way to upload chart packages. Using the same model of `Getter`, a new interface called `Pusher` should be introduced, with a single built-in implementation (OCI). This opens the door for plugins to implement their own upload mechanism, and for Helm to add a new top-level upload command `helm push`.
+
+### 2. Support for provenance files
+
+Helm currently has the ability to verify package signatures, assuming the presence of a file suffixed with `.prov` sitting next to a chart `.tgz` in a repository.
+
+Support for this method of signature validation should be carried over into OCI storage. As the format of the OCI manifest is custom to Helm, Helm can also choose to modify the resulting manifest on upload when a chart is being signed (e.g. `helm push --sign`).
+
+As far as the low-level details, the `.prov` file will simply be stored as a second layer on the manifest. If running `helm pull --verify oci://...`, the second layer (layer at index 1) will be assumed to be the provenance file. The first layer is always the chart `.tgz` itself.
+
+Here is an example of what a manifest will look like with a provenance file attached:
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "digest": "sha256:8ec7c0f2f6860037c19b54c3cfbab48d9b4b21b485a93d87b64690fdb68c2111",
+    "size": 117
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.cncf.helm.chart.v1",
+      "digest": "sha256:1b251d38cfe948dfc0a5745b7af5ca574ecb61e52aed10b19039db39af6e1617",
+      "size": 2487
+    },
+    {
+      "mediaType": "application/vnd.cncf.helm.provenance.v1",
+      "digest": "sha256:3e207b409db364b595ba862cdc12be96dcdad8e36c59a03b7b3b61c946a5741a",
+      "size": 643
+    }
+  ]
+}
+
+```
+
+### 3. Chart versions == OCI reference tags
+
+To keep things simple, the version of a chart will be 1-to-1 with the tag used on registry references. Arbitrary tags will not be supported.
+
+This also means that tags are no longer necessary to be provided on the command-line in the form `<ref>:<tag>`.
+
+The latest sematic version is able to be determined from a registry by listing all tags. To specify a specific tag/version (or if a registry does not support tag listing), users can provide `--version <version>`.
```

Comment:
As I'm reading this document, I'm looking for opportunities where we can concretely document how users should find this information.

Specifically:

> is able to be determined from a registry by listing all tags

How would a user list all tags with a given repository? Is there a helm command for that, or do we need to introduce a new `helm search registry` command to provide users with a solution?

Reply:
I've removed this for now.. We can address tag listing at a later point in time. `--version` will be required